internal/grpc/services/publicstorageprovider/publicstorageprovider.go

```diff
@@ -108,7 +108,25 @@ func New(m map[string]interface{}, ss *grpc.Server) (rgrpc.Service, error) {
 }
 
 func (s *service) SetArbitraryMetadata(ctx context.Context, req *provider.SetArbitraryMetadataRequest) (*provider.SetArbitraryMetadataResponse, error) {
-	return nil, gstatus.Errorf(codes.Unimplemented, "method not implemented")
+	tkn, relativePath, err := s.unwrap(ctx, req.Ref)
```

Comment:
Can you call `translatePublicRefToCS3Ref` instead?

Reply:
Good point. I changed that. :+1: